dataplane/src/pipeline/pipeline.rs

```diff
@@ -17,12 +17,27 @@ use crate::pipeline::{DynNetworkFunction, NetworkFunction, nf_dyn};
 #[derive(Default)]
 pub struct DynPipeline<Buf: PacketBufferMut> {
     nfs: Vec<Box<dyn DynNetworkFunction<Buf>>>,
+
+    #[allow(unused)]
+    name: String,
 }
 
+#[allow(unused)]
 impl<Buf: PacketBufferMut + 'static> DynPipeline<Buf> {
-    #[allow(unused)]
     pub fn new() -> Self {
-        Self { nfs: vec![] }
+        Self {
+            nfs: vec![],
+            name: "anonymous-pipeline".to_owned(),
+        }
+    }
+    pub fn with_name(name: &str) -> Self {
+        Self {
+            nfs: vec![],
+            name: name.to_owned(),
+        }
+    }
+    pub fn pipeline_name(&self) -> &str {
```

Comment:
Probably no need to keep `pipeline` in `pipeline_name()`, maybe just `name()` or `get_name()`?

Reply:
Yes, makes sense! Thanks @qmonnet !